src/Service/Startup.cs

```diff
@@ -324,6 +318,10 @@ private void AddGraphQLService(IServiceCollection services, GraphQLRuntimeOption
 
                     return error;
                 })
+                // Allows DAB to override the HTTP error code set by HotChocolate.
+                // This is used to ensure HTTP code 4XX is set when the datatbase
+                // returns a "bad request" error such as stored procedure params missing.
+                .UseRequest<DetermineStatusCodeMiddleware>()
```

Comment:
This middleware replaces the serializer. In HC 14 we can hint status codes we would want to use if the execution runs on the HTTP transport. This allows us to centralize the whole execution flow in one place.